torch_geometric/utils/metric.py

```diff
@@ -138,41 +58,14 @@ def f1_score(pred: Tensor, target: Tensor, num_classes: int) -> Tensor:
 
     :rtype: :class:`Tensor`
     """
-    prec = precision(pred, target, num_classes)
-    rec = recall(pred, target, num_classes)
-
-    score = 2 * (prec * rec) / (prec + rec)
-    score[torch.isnan(score)] = 0
-
-    return score
-
-
-def intersection_and_union(
-        pred: Tensor, target: Tensor, num_classes: int,
-        batch: Optional[Tensor] = None) -> Tuple[Tensor, Tensor]:
-    r"""Computes intersection and union of predictions.
-
-    Args:
-        pred (LongTensor): The predictions.
-        target (LongTensor): The targets.
-        num_classes (int): The number of classes.
-        batch (LongTensor): The assignment vector which maps each pred-target
-            pair to an example.
-
-    :rtype: (:class:`LongTensor`, :class:`LongTensor`)
-    """
-    pred, target = F.one_hot(pred, num_classes), F.one_hot(target, num_classes)
-
-    if batch is None:
-        i = (pred & target).sum(dim=0)
-        u = (pred | target).sum(dim=0)
-    else:
-        i = scatter_add(pred & target, batch, dim=0)
-        u = scatter_add(pred | target, batch, dim=0)
-
-    return i, u
 
 
+@deprecated(tm_func.jaccard_index, deprecated_in="v2.0", remove_in="v3.0",
+            args_mapping={
+                "pred": "preds",
+                "batch": None,
+                "omitnans": None
+            })
 def mean_iou(pred: Tensor, target: Tensor, num_classes: int,
              batch: Optional[Tensor] = None, omitnans: bool = False) -> Tensor:
```

Comment:
It looks like the `batch` argument support will be dropped here (mean `iou` across all examples). This might be a problem.

Reply:
ok, then we can do the batch over the IoU... but shall we do it while you suggest dropping metrics completely?

Reply:
What do you mean with doing the batch over the IoU?